test/model/compose_test.dart

```diff
@@ -314,6 +314,28 @@ hello
         '#narrow/dm/1,2-dm/near/12345',
         '#narrow/pm-with/1,2-pm/near/12345');
     });
+
+    test('normalize links to always include a "/" after hostname', () {
+      void checkGeneratedLink({required String realmUrl, required String expectedLink}) {
+        final account = eg.selfAccount.copyWith(realmUrl: Uri.parse(realmUrl));
+        final store = eg.store(account: account);
+        check(narrowLink(store, const CombinedFeedNarrow()))
+          .equals(Uri.parse(expectedLink));
```

Comment:
I'm slightly inclined to stringify the `narrowLink` result and compare that to the `String expectedLink`. The user's experience of a `narrowLink` result is always through `Uri.toString`. (Quote-and-reply and copy-link-to-message.)